lib/cloudregister/registerutils.py

```diff
@@ -173,6 +175,96 @@ def clear_rmt_as_scc_proxy_flag():
         pass
 
 
+# ----------------------------------------------------------------------------
+def clean_payg_extensions():
+    """Remove non free extensions files from /etc/products.d/."""
+    extensions = get_extensions()
+    products_paths = glob.glob('/etc/products.d/*.prod')
+
+    for extension in extensions:
+        # The following not free condition checks on
+        # the access capability of the extension
+        # This is a server side capability
+        if not extension.get('free', True):
+            extension_path = os.path.join(
+                os.sep, 'etc', 'products.d', '{}.prod'.format(
+                    extension.get('identifier')
+                )
+            )
+            if extension_path in products_paths:
+                os.unlink(extension_path)
+                logging.info(
+                    'Non free extension file %s removed' % extension_path
+                )
```

Comment:
We cannot just remove the `.prod` file. The product file was put there by an RPM package. By simply removing the prod file we put the system into an inconsistent state. The rpm database will say that the package that put the prod file there is installed but the file will be missing making it more difficult to understand any issues that may arise. We have to keep the system in a consistent state, meaning we have to remove the corresponding `-release` package that put the product file in place.

Should the user ever run an RPM DB consistency check they will get an error because of a problem we created, that is not an option.

Reply:
good catch

Reply:
yes this is true. I overlooked that in the review and testing.

```
registercloudguest --regcode ...

ls -l /etc/products.d/
total 16
-rw-r--r-- 1 root root 2679 Aug  1 12:52 SLES-LTSS.prod
...
```

As we only call SUSEConnect I thought that SLES-LTSS.prod was created by SUSEConnect, but in fact a package was installed

```
rpm -qf /etc/products.d/SLES-LTSS.prod
sles-ltss-release-15.4-150400.13.8.1.x86_64
```

ok, so yes we cannot simply remove the .prod file, thanks for catching this one.

It actually impacts some parts of the implementation.